src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -117,7 +117,17 @@ export function SavedObjectProvider(Promise, Private, Notifier, confirmModalProm
       try {
         searchSourceValues = JSON.parse(searchSourceJson);
       } catch (e) {
-        searchSourceValues = {};
+        throw new InvalidJSONProperty(
+          `Invalid JSON in ${esType} "${this.id}". ${e.message}`
```

Comment:
How about including `searchSourceJson` in error message to make it clear where InvalidJSONProperty is coming from

Reply:
I thought it would be a bit much, as this gets displayed in the UI, I think. But I agree it's handy for diagnostic purposes.